### ISSUE TITLE
Unable to use custom provider resources

### ISSUE THREAD

**Issue by nzmkey** [svdgraaf/serverless-plugin-stage-variables#3]
Serverless provide a way to add custom provider resources in serverless.yam.  EG:

``` yml
service: lambda-blah
custom:
  stageVariables:
    env: ${self:provider.stage} 
plugins:
  - serverless-plugin-stage-variables
provider: aws
functions:
  ...
resources:
  Resources:
    ApiGatewayStage:
      Type: AWS::ApiGateway::Stage
      Properties:
        MethodSettings:
          - DataTraceEnabled: true
            HttpMethod: "*"
            LoggingLevel: INFO
            ResourcePath: "/*"
            MetricsEnabled: true
```

The existing code will always overwrite the custom resources merged earlier in serverless/lib/plugins/aws/deploy/lib/mergeCustomProviderResources.js

Suggest do check if template.Resources.ApiGatewayStage already exists before setting

**Comment by nzmkey** [comment]
Working on the fix.  Will request pull soon.

**Comment by svdgraaf** [comment]
Thank you for the PR :+1:
I was way behind on github notifications, so only just found all the PR's! I merged your changed, and I just release v1.7.7, which includes your change!
